### DISCUSSION TITLE
User session dependency

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [alex-oleshkevich/starsessions#75] [Q&A]
Users make requests to a fastapi (or starlette) server during their session. A users request(n) payload data is stored so it is available to the next request(n+1) to extract information from. And, so on until the user exits their session. 

The server application does not have control over a users state in the client request (ie. cookie), only in the api. Imho, this means each user-session needs a unique session id to enable the payload data to be stored and accessed while the user-session is active.

Is this possible with starsessions?

**Comment by alex-oleshkevich** [comment]
I did not understand your use case. Each user has a unique session ID stored in the cookies. If you remove the cookie, the session will be regenerated and the previous data will be lost. So, if you cannot use cookies that sessions is a wrong component to solve your task.

**Reply by dineshbvadhia** [reply]
Typically, user state is handled by the client with cookies. In my case, developers build clients which access the fastapi (or starlette) server which doesn't have access to the cookie data ie. fastapi server <-> developer server <-> clients.

**Reply by alex-oleshkevich** [reply]
The session relies on cookies by design. IMO, you need some sort of database for this purpose.

**Reply by dineshbvadhia** [reply]
Hmmm. So, I need to know when a unique session has started, create a unique session id, store the payload data in a db, know when the session has ended, and delete the stored data for this user session ?

**Reply by alex-oleshkevich** [reply]
Session ID is generated during the response and stored in the cookie (sent in the response headers). Next request sends that cookie and sessions gets loaded by the ID from the backend. That's it.
